crates/turbopack-ecmascript/src/analyzer/mod.rs

```diff
@@ -2852,6 +2873,97 @@ impl WellKnownObjectKind {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct RequireContextOptions {
+    pub dir: String,
+    pub include_subdirs: bool,
+    /// this is a regex (pattern, flags)
+    pub filter: Regex,
+}
+
+pub fn regex_from_js(pattern: &str, flags: &str) -> Result<Regex, &'static str> {
```

Comment:
Would this always error if a JS regex is not representable as a Rust regex, or are there some inconsistencies between the two? e.g. some syntax that means something as a JS regex but a different thing as a rust regex.

Reply:
I think the "base" regex features are all the same and if anything there might be something that's supported only in rust or only in js